src/apps/myaccount/app.js

```diff
@@ -1021,6 +1017,55 @@ define(function(require) {
 					callback && callback(savedUser.data);
 				}
 			});
+		},
+
+		/**
+		 * Open balance tab and show add credit dialog
+		 */
+		showAddCreditDialog: function() {
+			var self = this;
+
+			self.showSubmodule({
+				module: 'balance',
+				callback: function() {
+					monster.pub('myaccount.balance.addCreditDialog');
+				}
+			});
+		},
+
+		/**
+		 * Displays the main view (if not displayed already), and activates the specified submodule
+		 * @param  {Object}   args
+		 * @param  {String}   args.module    Submodule name
+		 * @param  {Function} args.callback  Function called after the submodule has been activated
+		 */
+		showSubmodule: function(args) {
+			var self = this,
+				moduleName = args.module,
+				callback = args.callback;
+
+			monster.series([
+				function(seriesCallback) {
+					if ($(self.mainContainer).hasClass('myaccount-open')) {
+						seriesCallback(null);
+					} else {
+						self.renderDropdown(true, function() {
+							seriesCallback(null);
+						});
+					}
+				},
+				function(seriesCallback) {
+					self.activateSubmodule({
+						title: self.i18n.active()[moduleName].title,
+						module: moduleName,
+						callback: function() {
+							seriesCallback(null);
+						}
+					});
+				}
+			], function(err) {
+				!err && callback && callback();
```

Comment:
Let's not use selection operators in place of control statements.

We only tolerate `callback && callback()` because it is a legacy pattern.

```suggestion
				if (err) {
					return;
				}
				callback && callback();
```